bitcoind-rpc/src/main/scala/org/bitcoins/rpc/util/AsyncUtil.scala

```diff
@@ -97,7 +95,10 @@ abstract class AsyncUtil extends BitcoinSLogger {
         val p = Promise[Boolean]()
         val runnable = retryRunnable(condition, p)
 
-        system.scheduler.scheduleOnce(duration, runnable)
+        Future {
+          Thread.sleep(duration.toMillis)
```

Comment:
Isn't this blocking? I agree with your description that this change is likely what is causing problems in tests.

Reply:
(In the sense that it keeps Threads hostage)